### ISSUE TITLE
Restricted Access setting for authenticated users

### ISSUE THREAD

**Issue by atran** [ManifoldScholar/manifold#2849]
Let's create a backend setting that allows for, when the Restricted Access flag is enabled, only authenticated users can access those projects. @zdavis 's approach is below:

Start here: https://github.com/ManifoldScholar/manifold/blob/master/api/app/models/settings.rb#L19
api/app/models/settings.rb:19

Add a new default to Settings.instance.general . You don’t need a DB migration to do this, because this general property is a single DB json field. Once you add this, you can see it in the console with Settings.instance.general[:your_property_name]

Next, adjust the strong parameter validation to allow the new property through in API requests: https://github.com/ManifoldScholar/manifold/blob/master/api/app/controllers/concerns/validation.rb#L299

Update the settings form component to add the new field: https://github.com/ManifoldScholar/manifold/blob/master/client/src/backend/containers/settings/General.js

At that point, you should be able to see the setting in the backend, change it, save the settings, and see the new setting persist.

Then, it’s a matter of adjusting the fully_readable_by? method in the Project authorizer. When projects are fully_readable by a user, the user can read it’s texts and resources — having an entitlement to a project makes it fully_readable. You’ll add some logic to this method so that if your new setting is true and if user is not present, then that method returns false.

That should more or less be all you have to do — you’ll want to take some care in that authorizer method to make sure this new setting interacts with other access-related settings in a way that makes sense.

So, in short: 1) add a new default to the setting model, 2) update the validation concern to allow the new property in requests, 3) update the client component to include the new field, and 4) incorporate your new setting into project authorizer logic.

**Comment by zdavis** [comment]
@bot funding

**Comment by zdavis** [comment]
Thank you for taking the time to open this feature request. The Manifold team reviewed this issue during our bi-weekly meeting and the consensus is that this feature makes sense and is in keeping with our overall vision for the platform. That said, we don’t currently have funding available to support development of this feature. We’ve added a “needs funding” label to this feature request and closed it for now. This label signifies that should we find funding for this feature, or if the requestor wants to explore funding development, we’d be open to further discussion. It also means that we’re willing to accept pull requests that implement this feature.

_This was an automated message, but please don't hesitate to reply. Our team watches these issues closely and will respond as soon as we're able to!_

**Comment by zdavis** [comment]
Hey Anthony, I think you were considering taking a stab at this. If you get to a point where you have a PR to open against this issue, please don't hesitate to reopen it.

**Comment by atran** [comment]
@zdavis got it. I just hardcoded it for now, but I plan on implementing the CMS toggle soon. Thanks for the guidance here.
